src/themes/material/components/chip-list.component.ts

```diff
@@ -0,0 +1,122 @@
+import {Component, Input, OnInit, ChangeDetectorRef} from '@angular/core';
+import {AppError} from '@grid/core/infrastructure';
+
+interface IChipValue {
+	value: any;
+	index: number;
+}
+const enterKey = 13;
+
+@Component({
+	selector: 'q-grid-chips',
+	templateUrl: './chip-list.tpl.html'
+})
+export class ChipListComponent implements OnInit  {
+	constructor(private cdRef: ChangeDetectorRef) {
+		this._editMode = false;
+	}
+
+	@Input('cell-value') chipValues: any[];
+
+	private _values: IChipValue[];
+
+	private newValue: any;
+	private _editMode: boolean;
+	private editIndex: number = -1;
+	private editValue: any;
+
+	get chipList(): any[] {
+		return this._values.map((item) => item.value);
+	}
+	set chipList(values: any[]){
+		this._values = values.map((itemValue, itemIndex) => {
+			return <IChipValue> {
+				value: itemValue,
+				index: itemIndex
+			};
+		});
+	};
+
+	private editMode(index: number) {
+		return this._editMode && this.editIndex === index;
+	}
+
+	private removeItem(index: number) {
+		this.chipValues.splice(index, 1);
```

Comment:
please for remove, add etc. use Command pattern + shortcuts (u can find examples in grid)